prow/googlecloudbuild/client/client.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	cloudbuild "cloud.google.com/go/cloudbuild/apiv1/v2"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	cloudbuildpb "google.golang.org/genproto/googleapis/devtools/cloudbuild/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	// GCB automatically assigns random build ids to a new build, so it's not
+	// possible to associate a GCB build with a prow job by id without storing
+	// mapping somewhere else. Utilizing GCB tags for achieving this.
+
+	// ProwLabelSeparater formats labels key:val pairs into `{key} ::: {val}`
+	ProwLabelSeparater = " ::: "
+)
+
+// Operator is the interace that's highly recommended to be used by any package
+// that imports current package. Unit test can be carried out with a mock
+// implemented under fake of current package
+type Operator interface {
+	GetBuild(ctx context.Context, project, id string) (*cloudbuildpb.Build, error)
+	ListBuildsByTag(ctx context.Context, project string, tags []string) ([]*cloudbuildpb.Build, error)
+	CreateBuild(ctx context.Context, project string, bld *cloudbuildpb.Build) (*cloudbuildpb.Build, error)
+	CancelBuild(ctx context.Context, project, id string) (*cloudbuildpb.Build, error)
+}
+
+// ProwLabel formats labels key:val pairs into `{key} ::: {val}`
+func ProwLabel(key, val string) string {
+	return key + ProwLabelSeparater + val
+}
+
+// LabelFromTag trims label into key:val pairs. returns empty string as value if
+// the label is not formatted as prow label format
+func LabelFromTag(tag string) (key, val string) {
+	parts := strings.SplitN(tag, ProwLabelSeparater, 2)
+	if len(parts) < 2 {
+		return tag, ""
+	}
+	return parts[0], parts[1]
+}
+
+// GetProwLabels gets labels from cloud build struct, simulating k8s pods labels format
+// of map[string]string
+func GetProwLabels(bld *cloudbuildpb.Build) map[string]string {
+	res := make(map[string]string)
+	for _, tag := range bld.Tags {
+		key, val := LabelFromTag(tag)
+		if val != "" {
+			res[key] = val
+		}
+	}
+	return res
+}
+
+// Client wraps native cloudbuild client
+type Client struct {
+	interactor *cloudbuild.Client
+}
+
+// NewClient creates a new Client, with optinal credentialFile
```

Comment:
s/optinal/optional

Nit: Also, I've noticed that your comment sentences lack a period at the end of the sentence. For cleanliness it would help to add those (at least for all top-level comments).